src/app/forum/containers/thread/thread.component.ts

```diff
@@ -157,29 +157,41 @@ export class ThreadComponent implements AfterViewInit, OnDestroy {
     );
 
     this.subscriptions.add(
-      this.isThreadStatusOpened$.pipe(delay(0)).subscribe(isThreadStatusOpened => {
-        if (!isThreadStatusOpened) {
-          this.form.get('messageToSend')?.disable();
+      this.threadStatus$.pipe(delay(0)).subscribe(status => {
+        if (status?.data && (status.data.open || status.data.canOpen)) {
+          this.form.get('messageToSend')?.enable();
           return;
         }
-        this.form.get('messageToSend')?.enable();
-        this.messageToSendEl?.nativeElement.focus();
+        this.form.get('messageToSend')?.disable();
       })
     );
+    this.subscriptions.add(
+      this.isThreadStatusOpened$.pipe(delay(0), filter(isThreadStatusOpened => isThreadStatusOpened)).subscribe(() =>
+        this.messageToSendEl?.nativeElement.focus()
+      )
+    );
   }
 
   ngOnDestroy(): void {
     this.subscriptions.unsubscribe();
   }
 
-  sendMessage(threadId: ThreadId): void {
+  sendMessage(threadId: ThreadId, isThreadOpened: boolean): void {
     const messageToSend = this.form.value.messageToSend?.trim();
     if (!messageToSend) return;
     this.store.select(forum.selectThreadToken).pipe(
       take(1), // send on the current thread (if any) only
       filter(isNotUndefined),
     ).subscribe(token => this.forumWebsocketClient.send(publishEventsAction(token, [ messageEvent(messageToSend) ])));
-    this.updateThreadService.update(threadId.itemId, threadId.participantId, { messageCountIncrement: 1 }).subscribe();
+    if (isThreadOpened) {
+      this.updateThreadService.update(threadId.itemId, threadId.participantId, { messageCountIncrement: 1 }).subscribe();
+    } else {
+      this.changeThreadStatus({
+        open: true,
+        threadId,
+        messageCountIncrement: 1,
+      });
```

Comment:
I would say that it should open the thread first and then post the message... otherwise your "token" may even not allow you to post the message (I don't even know why it allows it? is it a freedom I left for now?).

Reply:
I don't even know why it allows it? is it a freedom I left for now?

> Yep

Reply:
Fixed